File: Data Structures & Algorithms/binary-search/submission-19.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        l = 0
        r = len(nums) - 1
        i = 0

        while l <= r:
            i += 1
            mid = (l + r) //2

            if nums[mid] == target:
                return mid
            elif nums[mid] < target:
                l = mid + 1
            else:
                r = mid - 1
        
        return -1

File: Data Structures & Algorithms/top-k-elements-in-list/submission-19.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} k
     * @return {number[]}
     */
    topKFrequent(nums, k) {
        let count = {}
        nums.forEach(val => (count[val] = (count[val] || 0) + 1))

        let freq = [[]]
        for(const num in nums){
            freq.push([])
        }

        for (const [val, occ] of Object.entries(count)){
            freq[occ].push(val)
        }

        let i = freq.length - 1
        let res = []
        while (i > 0){
            for (const item of freq[i]){
                res.push(item)
                if (res.length == k){
                    return res
                }
            }
            i --
        }
        return []
    }
}
